src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java

```diff
@@ -99,14 +100,118 @@ public String getVersion() {
 
+    /**
+     * Creates a {@link List} of {@link Funcotation} for the given {@code variant}, {@code referenceContext}, and {@code featureContext}.
+     * Accounts for override values passed into the constructor as well.
+     * @param variant {@link VariantContext} to annotate.
+     * @param referenceContext {@link ReferenceContext} corresponding to the given {@code variant}.
+     * @param featureSourceMap {@link Map} of {@link String} -> {@link List} of {@link Feature} (data source name -> data source features corresponding to the given {@code variant}.
+     * @return {@link List} of {@link Funcotation} given the {@code variant}, {@code referenceContext}, and {@code featureContext}.  This should never be empty.
+     */
+    public List<Funcotation> createFuncotations(final VariantContext variant, final ReferenceContext referenceContext, final Map<String, List<Feature>> featureSourceMap) {
+
+        // Get the features that this funcotation factory is responsible for:
+        final List<Feature> featureList = getFeatureListFromMap(featureSourceMap);
+
+        // If our featureList is compatible with this DataSourceFuncotationFactory, then we make our funcotations:
+        if ( isFeatureListCompatible(featureList) ) {
+
+            // Create our funcotations:
+            final List<Funcotation> outputFuncotations = createFuncotationsOnVariant(variant, referenceContext, featureList);
+
+            // Set our overrides:
+            setOverrideValuesInFuncotations(outputFuncotations);
+
+            return outputFuncotations;
+        }
+        else {
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Creates a {@link List} of {@link Funcotation} for the given {@code variant}, {@code referenceContext}, {@code featureContext}, and {@code gencodeFuncotations}.
+     * For some Data Sources knowledge of Gene Name or Transcript ID is required for annotation.
+     * Accounts for override values passed into the constructor as well.
+     * @param variant {@link VariantContext} to annotate.
+     * @param referenceContext {@link ReferenceContext} corresponding to the given {@code variant}.
+     * @param featureSourceMap {@link Map} of {@link String} -> {@link List} of {@link Feature} (data source name -> data source features corresponding to the given {@code variant}.
+     * @param gencodeFuncotations {@link List} of {@link GencodeFuncotation} that have already been created for the given {@code variant}/{@code referenceContext}/{@code featureContext}.
+     * @return {@link List} of {@link Funcotation} given the {@code variant}, {@code referenceContext}, and {@code featureContext}.  This should never be empty.
+     */
+    public List<Funcotation> createFuncotations(final VariantContext variant, final ReferenceContext referenceContext, final Map<String, List<Feature>> featureSourceMap, final List<GencodeFuncotation> gencodeFuncotations) {
```

Comment:
Can you combine this method with the one above to leverage shared code?  If a data source needs gencode info, but `createFuncotation(...)` is called without Gencode, will an exception be thrown?

Reply:
Ha!  Yeah - I forgot to combine them.

Yeah - I'll add that in.

Reply:
Yeah - the exception gets thrown in the concrete classes implementations of `createFuncotations`